tests/network/sriov.go

```diff
@@ -689,6 +707,33 @@ func waitVMI(vmi *v1.VirtualMachineInstance) (*v1.VirtualMachineInstance, error)
 	return virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
 }
 
+// deleteVMI deletes the specified VMI and waits for its absence.
+// Waiting for the VMI removal is placed intentionally for VMI/s with SR-IOV networks in order
+// to assure resources (VF/s) are fully released before reused again on a new VMI.
+func deleteVMI(vmi *v1.VirtualMachineInstance) error {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		panic(err)
+	}
+
+	GinkgoWriter.Println("sriov:", vmi.Name, "deletion started")
+	if err = virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &k8smetav1.DeleteOptions{}); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to delete VMI: %v", err)
+	}
+
+	const timeout = 30 * time.Second
+	return wait.PollImmediate(1*time.Second, timeout, func() (done bool, err error) {
```

Comment:
Please reuse `utils.WaitForVirtualMachineToDisappearWithTimeout`.

Reply:
If this is the case, you can move this helper to a more proper package and add offset parameter to it.
It doesn't justify a copy paste.

Reply:
It is a different approach, not a copy/paste.
One asserts in helpers, the other is pure functional and returns errors.
One depends on ginkgo & gomega, the other does not.
One is dependent on offsets, the other does not.

Are you asking to refactor `WaitForVirtualMachineToDisappearWithTimeout` or extracting with "wait" implementation to a function and placing it in a new package?

Reply:
> Are you asking to refactor WaitForVirtualMachineToDisappearWithTimeout or extracting the "wait.Poll*" implementation to a function and placing it in a new package?

I"m o with both of the options.

Reply:
I opened a follow up PR for this: https://github.com/kubevirt/kubevirt/pull/8365